### PR TITLE
feat: add memcached go sample

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
Add Memcached Go sample

**Motivation for the change:**

operator-framework/enhancements#47

### CODE REVIEW

**Comment by camilamacedo86** [comment]
/hold 

Let's wait the https://github.com/operator-framework/operator-sdk/pull/3981 get merged for we apply the same suggestions 👍

**Review by jmrodri — APPROVED** [review]
/lgtm

File: hack/generate/samples/go/mencached_with_webooks.go [hunk 1]
```diff
@@ -0,0 +1,443 @@
+// Copyright 2020 The Operator-SDK Authors
```
Comment (joelanford) [inline hack/generate/samples/go/mencached_with_webooks.go:1]: Fix typos in filename `s/mencached/memcached/` and `s/webooks/webhooks/`

**Comment by joelanford** [comment]
Just merged #3993, which caused some conflicts. Make sure to move `./hack/generate/samples/go` to `./hack/generate/samples/internal/go`

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

All done. would you like to give a look for we get it merged?

**Review by joelanford — APPROVED** [review]
/lgtm

**Pull request merged**